### PR TITLE
Several improvements to mapping

### PR DESCRIPTION
**Pull request by samleeflang**
Bunch of improvements for the mapping.
Noticed several issues with ABCD(EFG):
- coordinates are double and weren't picked up
- modified time was in long and wasn't picked up

Fixed differences with David's mapping.
https://naturalis.atlassian.net/browse/DD-445
- Check for modified in metadata
- Check for license in unit data
- Check more fields for specimen name

Able to retrieve tax info for identification extension:
https://naturalis.atlassian.net/browse/DD-495
- When no tax info in core file check extension
- When there is an identificationStatus only take ones that have "1" (ignore others)
- When no identificationStatus take first identification

### CODE REVIEW

File: src/main/java/eu/dissco/core/translator/terms/License.java [hunk 1]
```diff
@@ -10,28 +9,24 @@ public class License extends Term {
 
   public static final String TERM = "dcterms:license";
   private final List<String> dwcaTerms = List.of(TERM, "dc:license");
+  private final List<String> abcdUnitTerms = List.of("abcd:iprstatements/licenses/license/0/uri",
+      "abcd:iprstatements/licenses/license/0/text");
+  private final List<String> abcdMetaTerms = List.of(
+      "abcd:metadata/iprstatements/licenses/license/0/uri",
+      "abcd:metadata/iprstatements/licenses/license/0/text");
```
Comment (southeo) [inline src/main/java/eu/dissco/core/translator/terms/License.java:16]: Can you combine these lists? Or do you want to make sure Unit terms take precedence over meta terms?

File: src/main/java/eu/dissco/core/translator/terms/License.java [hunk 2]
```diff
@@ -10,28 +9,24 @@ public class License extends Term {
...
-        license = datasets.getMetadata().getIPRStatements().getLicenses().getLicense().get(0)
-            .getText();
-      }
-      return license;
+  public String retrieveFromABCD(JsonNode dataset, JsonNode unit) {
+    var license = searchJsonForStringTerm(unit, abcdUnitTerms);
+    if (license == null) {
+      license = searchJsonForStringTerm(dataset, abcdMetaTerms);
     }
-    log.info("Unable to get license for dataset");
```
Comment (southeo) [inline src/main/java/eu/dissco/core/translator/terms/License.java:33]: Is it possible there's nothing found when the meta terms are searched as well? Maybe this log should stay

File: src/main/java/eu/dissco/core/translator/terms/specimen/taxonomy/TaxonRank.java [hunk 3]
```diff
@@ -11,9 +11,11 @@ public class TaxonRank extends AbstractTaxonomy {
 
   @Override
   public String retrieveFromDWCA(JsonNode unit) {
-    return super.searchJsonForTerm(unit, dwcaTerms);
+    return super.getTaxonFromDWCA(unit, dwcaTerms);
   }
 
+
```
Comment (southeo) [inline src/main/java/eu/dissco/core/translator/terms/specimen/taxonomy/TaxonRank.java:17]: remove spaces

**Review by southeo — APPROVED** [review]
Looks good! Few clarifying questions but nothing blocking. 
That Identification index is a doozy...

File: src/main/java/eu/dissco/core/translator/terms/specimen/taxonomy/TaxonRank.java (hunk 3 above)
Comment (samleeflang) [reply src/main/java/eu/dissco/core/translator/terms/specimen/taxonomy/TaxonRank.java:17]: Yes thanks, ran a reformat over everything

File: src/main/java/eu/dissco/core/translator/terms/License.java (hunk 1 above)
Comment (samleeflang) [reply src/main/java/eu/dissco/core/translator/terms/License.java:16]: Yes, the path is different and the object is searches over is also different, so it is hard to combine these two lists.

File: src/main/java/eu/dissco/core/translator/terms/License.java (hunk 2 above)
Comment (samleeflang) [reply src/main/java/eu/dissco/core/translator/terms/License.java:33]: Not sure about this. If we do it, we might do it everywhere. Lots of times the license is missing so it gives a lot of logging. We could put it on debug but then we should do it with all the terms.

**Comment by samleeflang** [comment]
Yes so the identificationIndex was tricky. What I didn't wanted to do was recheck the identificationIndex for each property. So approached it a bit like a singleton. The first time something is called that needs in identificationIndex we calculate it and store it in a temporary property (in the original data). The next times we first check the temp property, which is now filled, and use the value from there. Before we commit the orginal data we remove the temp property. Bit trick but it means we don't unnecessary loop over all the elements.

**Comment by samleeflang** [comment]
Sonar fails on duplication, but further abstraction is not possible. These are just one liners pointing to super methods. By keeping them in separate files means we will still have the flexibility to include custom logic to the process.

**Review by southeo — APPROVED** [review]
Looks good

**Pull request merged**